flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/row/ParquetRowDataWriter.java

```diff
@@ -224,6 +293,176 @@ private TimestampWriter(int precision) {
         public void write(RowData row, int ordinal) {
             recordConsumer.addBinary(timestampToInt96(row.getTimestamp(ordinal, precision)));
         }
+
+        @Override
+        public void write(Object value) {
+            recordConsumer.addBinary(timestampToInt96((TimestampData) value));
+        }
+    }
+
+    private class MapWriter implements FieldWriter {
+
+        private String repeatedGroupName;
+        private String keyName, valueName;
+        private FieldWriter keyWriter, valueWriter;
+        private ArrayData.ElementGetter keyElementGetter, valueElementGetter;
+
+        private MapWriter(LogicalType keyType, LogicalType valueType, GroupType groupType) {
+
+            // Get the internal map structure (MAP_KEY_VALUE)
+            GroupType repeatedType = groupType.getType(0).asGroupType();
+            this.repeatedGroupName = repeatedType.getName();
+
+            // Get key element information
+            Type type = repeatedType.getType(0);
+            this.keyName = type.getName();
+            this.keyWriter = createWriter(keyType, type);
+
+            // Get value element information
+            Type valuetype = repeatedType.getType(1);
+            this.valueName = valuetype.getName();
+            this.valueWriter = createWriter(valueType, valuetype);
+
+            this.keyElementGetter = ArrayData.createElementGetter(keyType);
+            this.valueElementGetter = ArrayData.createElementGetter(valueType);
+        }
+
+        @Override
+        public void write(RowData row, int ordinal) {
+            recordConsumer.startGroup();
+
+            MapData mapData = row.getMap(ordinal);
+
+            if (mapData != null && mapData.size() > 0) {
+                recordConsumer.startField(repeatedGroupName, 0);
+
+                ArrayData keyArray = mapData.keyArray();
+                ArrayData valueArray = mapData.valueArray();
+                for (int i = 0; i < keyArray.size(); i++) {
+                    Object key = keyElementGetter.getElementOrNull(keyArray, i);
+                    Object value = valueElementGetter.getElementOrNull(valueArray, i);
+
+                    recordConsumer.startGroup();
+                    if (key != null) {
```

Comment:
Key is nullable?

Reply:
supports multiple types in latest commit.

Reply:
supports null in latest commit